operator/deploy/crd.yaml

```diff
@@ -30,11 +30,335 @@ spec:
+                        ldap_member_of_key:
+                          type: string
+                          default: "memberOf"
+                        ldap_port:
+                          type: number
+                        ldap_role_filter:
+                          type: string
+                        ldap_search_filter:
+                          type: string
+                        ldap_use_ssl:
+                          type: boolean
+                        ldap_user_filter:
+                          type: string
+                        ldap_user_id_key:
+                          type: string
+                          default: "cn"
+                deployment:
+                  type: object
+                  properties:
+                    accessible_namespaces:
+                      type: array
+                      items:
+                        type: string
+                    additional_service_yaml:
+                      type: object
+                      additionalProperties: true
+                    affinity:
+                      type: object
+                      properties:
+                        affinity:
+                          type: object
+                          additionalProperties: true
+                        pod:
+                          type: object
+                          additionalProperties: true
+                        pod_anti:
+                          type: object
+                          additionalProperties: true
+                    image_name:
+                      type: string
+                    image_pull_policy:
+                      type: string
+                      enum:
+                        - Always
+                        - Never
+                        - IfNotPresent
+                    image_pull_secrets:
+                      type: array
+                      items:
+                        type: string
+                    image_version:
+                      type: string
+                    ingress_enabled:
+                      type: boolean
+                      default: true
+                    namespace:
+                      type: string
+                    secret_name:
+                      type: string
+                    service_type:
+                      type: string
+                      enum:
+                        - NodePort
+                        - ClusterIP
+                        - LoadBalancer
+                        - ExternalName
+                    tolerations:
+                      type: array
+                      items:
+                        type: object
+                        additionalProperties: true
+                    verbose_mode:
+                      type: number
+                      minimum: 0
+                      maximum: 5
+                    version_label:
+                      type: string
+                    view_only_mode:
+                      type: boolean
```

Comment:
if we ARE going to put defaults in this schema, then we need to set this default which is `false`

(see what a pain it is to keep defaults in sync here :)) I won't continue with these comments - suffice it to say, there are alot of defaults missing in here. Either we document them all, or we do not document any defaults. But we shouldn't document just some of them)